File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/aop/support/AnnotationClassOrMethodPointcut.java
package org.xyattic.eventual.consistency.support.core.aop.support;

import org.springframework.aop.ClassFilter;
import org.springframework.aop.MethodMatcher;
import org.springframework.aop.Pointcut;

import java.lang.annotation.Annotation;

/**
 * @author wangxing
 * @create 2020/11/12
 */
public class AnnotationClassOrMethodPointcut implements Pointcut {

    private final MethodMatcher methodMatcher;


    public AnnotationClassOrMethodPointcut(Class<? extends Annotation> annotationType) {
        this.methodMatcher = new AnnotationClassOrMethodMatcher(annotationType);
    }

    @Override
    public ClassFilter getClassFilter() {
        return ClassFilter.TRUE;
    }

    @Override
    public MethodMatcher getMethodMatcher() {
        return methodMatcher;
    }

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/aop/support/AnnotationClassOrMethodPointcut.kt
package org.xyattic.eventual.consistency.support.core.aop.support

import org.springframework.aop.ClassFilter
import org.springframework.aop.MethodMatcher
import org.springframework.aop.Pointcut

/**
 * @author wangxing
 * @create 2020/11/12
 */
open class AnnotationClassOrMethodPointcut(annotationType: Class<out Annotation>) : Pointcut {

    private val methodMatcher: MethodMatcher

    override fun getClassFilter(): ClassFilter {
        return ClassFilter.TRUE
    }

    override fun getMethodMatcher(): MethodMatcher {
        return methodMatcher
    }

    init {
        methodMatcher = AnnotationClassOrMethodMatcher(annotationType)
    }
}